### PR TITLE
Add clearDatastore flag

### PR DESCRIPTION
**Pull request by loosebazooka**
Also fix some minor checkstyle issue

### CODE REVIEW

**Comment by loosebazooka** [comment]
Fixes #303

File: src/main/java/com/google/cloud/tools/appengine/api/devserver/RunConfiguration.java [hunk 1]
```diff
@@ -72,4 +72,6 @@
   String getDefaultGcsBucketName();
 
   String getJavaHomeDir();
+
+  Boolean getClearDatastore();
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/api/devserver/RunConfiguration.java]: 1. Needs Javadoc.
2. Effective Java recommends using boolean primitive

File: src/main/java/com/google/cloud/tools/appengine/api/devserver/DefaultRunConfiguration.java [hunk 2]
```diff
@@ -254,4 +255,13 @@ public String getJavaHomeDir() {
   public void setJavaHomeDir(String javaHomeDir) {
     this.javaHomeDir = javaHomeDir;
   }
+
+  @Override
+  public Boolean getClearDatastore() {
+    return clearDatastore;
+  }
+
+  public void setClearDatastore(Boolean clearDatastore) {
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/api/devserver/DefaultRunConfiguration.java]: needs Javadoc

File: src/main/java/com/google/cloud/tools/appengine/api/devserver/RunConfiguration.java (hunk 1 above)
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/api/devserver/RunConfiguration.java]: We want default behavior to be null, so we always delegate to gcloud defaults (which can be true or false)

**Comment by patflynn** [comment]
@loosebazooka 
Are you thinking of introducing the configuration property independently of plumbing it into the run command?

**Comment by loosebazooka** [comment]
no, looks like I forgot to add a file in my rush to eat lunch... :open_mouth:

File: src/main/java/com/google/cloud/tools/appengine/api/devserver/DefaultRunConfiguration.java (hunk 2 above)
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/api/devserver/DefaultRunConfiguration.java]: We haven't been documenting value class getters/setters like this in this project. Do you think we should?

Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/api/devserver/DefaultRunConfiguration.java]: Yes. This needs multiple things clearly stated:

1. What this flag does. i.e. mostly copy the gcloud docs on this
2. What the default is and what null means.

**Pull request merged**